extension/src/experiments/model/index.ts

```diff
@@ -857,22 +857,34 @@ export class ExperimentsModel extends ModelWithPersistence {
     return commitsBySha
   }
 
+  private addBranchToFlattenedCommit(commitAndExps: Commit[], branch: string) {
+    return commitAndExps.map(commitOrExp => ({
+      ...commitOrExp,
+      branch: commitOrExp.branch || branch,
+      otherBranches: commitOrExp?.otherBranches
+        ? [...commitOrExp.otherBranches, branch]
+        : []
+    }))
+  }
+
   private getFlattenedRowData(workspaceRow: Commit): Commit[] {
     const commitsBySha: { [sha: string]: Commit[] } =
       this.applyFiltersToFlattenedCommits()
-    const rows = []
+    const rowsBySha: { [sha: string]: Commit[] } = {}
```

Comment:
[I] This would be easier to follow if you loop over `rowOrder` to collapse the branches into `{ [sha]: string[]}` and then apply those to `Object.values(commitsBySha)`

Reply:
Good idea!